### ISSUE TITLE
Compile /w cython

### ISSUE THREAD

**Issue by George3d6** [mindsdb/lightwood#476]
After #473 is done

Try compiling with cython and see if that improves performance (realistically it will fail to compile and once we put 1-2 hours into trying to get it to we should give up and shelve this for when we have more time)

**Comment by George3d6** [comment]
Leaving this on the backburner, seems like this is surprisingly enough viable... but it will make our builds hell: https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#configuring-the-c-build

And I'm not sure if the resulting monsters won't require various shared libraries.

The main idea is something like:

1. Have a step where we compile x/y/z.py files that are slow with cython
2. Call them via C API where they are required
3. Include them in the dist of the package

We can't compile the library as a whole with cython, and it seems best to leave the top level API in python.

But this method would involve a heavy workload and introduce a lot of complexity, since we'd have to *both* change the lightwood code to import C function *and* we'd have to setup a lengthy build process that might require a somewhat specific env. That is assuming the C code we compile works on every machine, and we don't need to compile separately for every single dist we want to support... which is likely what we'd need to do.

So overall will postpone until we have a larger team and/or more time.
